### PR TITLE
add shepp_logan image

### PR DESCRIPTION
**Pull request by johnnychen94**
Move `shepp_logan` function here from Images.jl  https://github.com/JuliaImages/Images.jl/issues/899

Changes:

* the previous implementation returns `Array{Float64, 2}`, this one returns `Array{Gray{Float64}, 2}`
* rename keyword argument `highContrast` to `high_contrast`; the previous one follows MATLAB style.
* 1~2x speed up......

```julia
# Julia v1.5.0-rc1
julia> @btime Images.shepp_logan(512);
  25.400 ms (290 allocations: 122.53 MiB)

julia> @btime TestImages.shepp_logan(512);
  15.213 ms (72 allocations: 4.01 MiB)

julia> @btime Images.shepp_logan(128);
  1.207 ms (280 allocations: 7.74 MiB)

julia> @btime TestImages.shepp_logan(128);
  916.064 μs (72 allocations: 261.19 KiB)
```

Question: would it be convenient to expose this to `testimage`? For example, `testimage("phantom", (512, 512))`

### CODE REVIEW

File: src/TestImages.jl [hunk 1]
```diff
@@ -113,12 +114,60 @@ end
...
+        grayLevel = [2, -0.98, -0.02, -0.02, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
+    end
+  
+    P = zeros(N, M)
+    for l = 1:length(theta)
+      P .+= grayLevel[l] .* (
+             abs2.( (cos(theta[l]) .* (x .- centerX[l]) .+
+                sin(theta[l]) .* (y .- centerY[l])) ./ majorAxis[l] ) .+
+             abs2.( (sin(theta[l]) .* (x .- centerX[l]) .-
+                cos(theta[l]) .* (y .- centerY[l])) ./ minorAxis[l] ) .< 1 )
```
Comment (johnnychen94) [inline src/TestImages.jl:166]: I thought it is equivalent, but benchmark tells me that

```julia
      @. P += grayLevel[l] * (
             abs2( (cos(theta[l]) * (x - centerX[l]) +
                sin(theta[l]) * (y - centerY[l])) / majorAxis[l] ) +
             abs2( (sin(theta[l]) * (x - centerX[l]) -
                cos(theta[l]) * (y - centerY[l])) / minorAxis[l] ) < 1 )
```

is ~2x slower than the current version.

Comment (kimikage) [reply src/TestImages.jl:169]: BTW, we can avoid the reallocation.
```julia
P = zeros(Gray{Float64}, N, M)
    for l = 1:length(theta)
      P .= gray.(P) .+ grayLevel[l] .* (
...
```
Probably no slowdown, but no speedup.

Comment (johnnychen94) [reply src/TestImages.jl:169]: Yes, I'm aware of this. Just that I prefer to not depend on `ColorVectorSpace`.

Comment (kimikage) [reply src/TestImages.jl:169]: ?
```julia
julia> using ColorTypes

julia> P = zeros(Gray{Float64}, 2, 2);

julia> P .= 1.0
2×2 Array{Gray{Float64},2} with eltype Gray{Float64}:
 Gray{Float64}(1.0)  Gray{Float64}(1.0)
 Gray{Float64}(1.0)  Gray{Float64}(1.0)
```

Comment (johnnychen94) [reply src/TestImages.jl]: 😂 I was too careless that I misunderstood it!

Comment (kimikage) [reply src/TestImages.jl]: I know this PR isn't about optimization, but I think it's easier to read using functions.
```julia
function shepp_logan(N::Integer, M::Integer; high_contrast=true)
    x = range(-1, stop=1, length=M)'
    y = range(1, stop=-1, length=N)
  
    center_x = (0.0, 0.0, 0.22, -0.22, 0.0, 0.0, 0.0, -0.08, 0.0, 0.06)
    center_y = (0.0, -0.0184, 0.0, 0.0, 0.35, 0.1, -0.1, -0.605, -0.605, -0.605)
    major_axis = (0.69, 0.6624, 0.11, 0.16, 0.21, 0.046, 0.046, 0.046, 0.023, 0.023)
    minor_axis = (0.92, 0.874, 0.31, 0.41, 0.25, 0.046, 0.046, 0.023, 0.023, 0.046)
    theta = (0.0, 0.0, -0.1pi, 0.1pi, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
  
    if high_contrast
        # high contrast (MRI) version of the phantom -- [2] p.201
        gray_level = (1.0, -0.8, -0.2, -0.2, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1)
    else
        # original (CT) version of the phantom -- [1]
        gray_level = (2.0, -0.98, -0.02, -0.02, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01)
        # [3] p.439 uses the following setting, and is what MATLAB's built-in `phantom` uses
        # gray_level = (1.0, -0.98, -0.02, -0.02, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01)
    end
    
    function ellipse(dx, dy, a, b, cos_t, sin_t)
        tx = cos_t * dx + sin_t * dy
        ty = sin_t * dx - cos_t * dy
        abs2(tx * b) + abs2(ty * a) < (a * b)^2
    end
    function ellipse(dx, dy, a, b)
        abs2(dx * b) + abs2(dy * a) < (a * b)^2
    end
  
    P = zeros(Gray{Float64}, N, M)
    for l = 1:length(theta)
        dx = x .- center_x[l]
        dy = y .- center_y[l]
        if theta[l] == 0.0
            P .= gray.(P) .+ gray_level[l] .* ellipse.(dx, dy, major_axis[l], minor_axis[l])
        else
            P .= gray.(P) .+ gray_level[l] .* 
                ellipse.(dx, dy, major_axis[l], minor_axis[l], cos(theta[l]), sin(theta[l]))
        end
    end

    return P
end
```

**Comment by johnnychen94** [comment]
After e9d7643:

```julia
julia> @btime TestImages.shepp_logan(512);
  3.752 ms (4 allocations: 2.01 MiB)
```

Thanks for the suggestion @kimikage

**Pull request merged**